manta-benchmark/src/payment.rs

```diff
@@ -0,0 +1,254 @@
+    assert_valid_proof(verifying_context, &mint);
+}
+
+/// Samples num [`Mint`]s
+#[inline]
+pub fn sample_num_mints<R>(
+    num: usize,
+    proving_context: &MultiProvingContext,
+    verifying_context: &MultiVerifyingContext,
+    parameters: &Parameters,
+    utxo_accumulator_model: &UtxoAccumulatorModel,
+    assets: &[Asset],
+    rng: &mut R,
+) -> (
+    UtxoAccumulator,
+    Vec<config::SpendingKey>,
+    Vec<config::Sender>,
+)
+where
+    R: CryptoRng + RngCore + ?Sized,
+{
+    let mut utxo_accumulator = UtxoAccumulator::new(utxo_accumulator_model.clone());
+    let mut vec_spending_keys = Vec::with_capacity(num);
+    let mut vec_senders = Vec::with_capacity(num);
+
+    for i in 0..num {
+        let spending_key = SpendingKey::new(rng.gen(), rng.gen());
+        let (mint, pre_sender) = transfer::test::sample_mint(
+            &proving_context.mint,
+            FullParameters::new(parameters, utxo_accumulator.model()),
+            &spending_key,
+            assets[i],
+            rng,
+        )
+        .expect("Unable to build MINT proof.");
+        assert_valid_proof(&verifying_context.mint, &mint);
+        let sender = pre_sender
+            .insert_and_upgrade(&mut utxo_accumulator)
+            .expect("Just inserted so this should not fail.");
+
+        vec_spending_keys[i] = spending_key;
+        vec_senders[i] = sender;
+    }
+
+    (utxo_accumulator, vec_spending_keys, vec_senders)
+}
+
+/// a wrapper for 2 mints and 1 private transfer
+#[inline]
+pub fn bench_private_transfer_wrapper<R>(
+    proving_context: &MultiProvingContext,
+    verifying_context: &MultiVerifyingContext,
+    parameters: &Parameters,
+    utxo_accumulator_model: &UtxoAccumulatorModel,
+    assets: Vec<Asset>,
+    rng: &mut R,
+) where
+    R: CryptoRng + RngCore + ?Sized,
+{
+    let mut utxo_accumulator = UtxoAccumulator::new(utxo_accumulator_model.clone());
+
+    let spending_key_0 = SpendingKey::new(rng.gen(), rng.gen());
+    let (mint_0, pre_sender_0) = transfer::test::sample_mint(
+        &proving_context.mint,
+        FullParameters::new(parameters, utxo_accumulator.model()),
+        &spending_key_0,
+        assets[0],
+        rng,
+    )
+    .expect("Unable to build MINT proof.");
+    assert_valid_proof(&verifying_context.mint, &mint_0);
+    let sender_0 = pre_sender_0
+        .insert_and_upgrade(&mut utxo_accumulator)
+        .expect("Just inserted so this should not fail.");
+
+    let spending_key_1 = SpendingKey::new(rng.gen(), rng.gen());
+    let (mint_1, pre_sender_1) = transfer::test::sample_mint(
+        &proving_context.mint,
+        FullParameters::new(parameters, utxo_accumulator.model()),
+        &spending_key_1,
+        assets[1],
+        rng,
+    )
+    .expect("Unable to build MINT proof.");
+    assert_valid_proof(&verifying_context.mint, &mint_1);
+    let sender_1 = pre_sender_1
+        .insert_and_upgrade(&mut utxo_accumulator)
+        .expect("Just inserted so this should not fail.");
```

Comment:
Need to extract these parts out of the benchmark-able code and try to use `sample_mints`.